index.js

```diff
@@ -121,6 +128,7 @@ module.exports = class Filter extends Plugin {
       loggerName += ' > [' + annotation + ']';
     }
 
+    this.fsMerger = new FSMerger(inputTrees);
```

Comment:
if this is a subclass of broccoli-plugin, and broccoli-plugin has `this.input` and `this.output` why do we need `FSMerger` here?

Reply:
Because we want to support 
https://github.com/broccolijs/broccoli-persistent-filter/blob/9e01bdae7da04f6f7146c4dee1818d2fe75e1339/test/test.js#L1524 case.
`this.input` is a `broccoli-plugin` feature which expects `this.inputNodes` to be broccoli node or a string.

Reply:
Was that mistake, should this have been a primitive features of `this.input`? Or should the code in question operate on `this.input` rather then `inputTrees` directly?

Currently I am worried, that this plugin does not use `this.input`, did we make a mistake with it? By not using `this.input` we are making it hard for future optimizations, as ensuring all plugins go through `this.input` `this.output` opens many doors.

Reply:
Should we consider handling this case separately and use input/output facade changes first and then see what we can to extend that feature?

Reply:
Unfortunately mixing input + the custom fsMerger object, will break encapsulation for subclasses of filter. Are they supposed to read from `this.input` or from the other?

Our goal here is, to ensure `this.input` is the usable for subclasses, as that is our new public API for reading from input trees.